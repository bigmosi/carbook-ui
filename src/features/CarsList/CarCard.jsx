/* eslint-disable camelcase */
/* eslint-disable react/prop-types */
import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './CarCard.module.css';

const Car = ({
  id, model, image, brand, price,
}) => (
  <NavLink className={styles.links} key={id} to={`/car/${id}`}>
    <div className={styles.carCnt} key={id}>
      <div className={styles.imgCnt}>
        <img src={image} alt={brand} className={styles.carImg} />
      </div>
      <div className="model-detail d-flex flex-column justify-content-center align-items-center">
        <h3 className="h4 mt-3">{brand}</h3>
        <div className="d-flex flex-row justify-content-center align-items-center gap-2 mt-2">
          <h5 className="h6 pt-1">{model}</h5>
        </div>
        <h5 className={styles.price}>
          $
          {' '}
          {price}
        </h5>
      </div>
    </div>
  </NavLink>
);

export default Car;
