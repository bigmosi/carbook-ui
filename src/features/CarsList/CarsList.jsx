import React, { useEffect, useState } from 'react';
import axios from 'axios';
import Carousel from 'nuka-carousel/lib/carousel';
import { useTheme } from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';
import CarCard from './CarCard';
import './Carousel.css';

const CarList = () => {
  const [cars, setCars] = useState([]);
  const [loading, setLoading] = useState(false);
  useEffect(() => {
    setLoading(true);
    axios
      .get('http://localhost:3000/api/v1/cars')
      .then((response) => {
        setCars(response.data.cars);
        setLoading(false);
      });
  }, []);

  const theme = useTheme();
  const mdUp = useMediaQuery(theme.breakpoints.up('md'));
  const lgUp = useMediaQuery(theme.breakpoints.up('lg'));

  let SlideToShowNumber = 1;
  if (mdUp) {
    SlideToShowNumber = 2;
  }
  if (lgUp) {
    SlideToShowNumber = 3;
  }

  
  );
};

export default CarList;
