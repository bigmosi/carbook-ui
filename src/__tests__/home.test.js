import React from 'react';
import '@testing-library/jest-dom';
import renderWithProviders from '../utils/test-utils';
import CarsList from '../features/CarsList/CarsList';

describe('Homepage component', () => {
  const { asFragment } = renderWithProviders(<CarsList />);
  const firstRender = asFragment();

  it('should render correctly', () => {
    expect(firstRender).toMatchSnapshot();
  });
  it('should render correctly after state change', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
