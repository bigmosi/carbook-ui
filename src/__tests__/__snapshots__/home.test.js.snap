// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="container-fluid vh-100 v-100 d-flex justify-content-center align-items-center"
  >
    <i
      class="fa-solid fa-spinner fa-spin fs-1"
    />
  </div>
</DocumentFragment>
`;

exports[`Homepage component should render correctly after state change 1`] = `<DocumentFragment />`;
